### ISSUE TITLE
[otbn] Create top-level wrapper

### ISSUE THREAD

**Issue by imphil** [lowRISC/opentitan#2468]
Create top-level OTBN SV file with RAMs, interface definitions, etc.

The external interfaces are also part of the specification PR #2393 and may need to be copied into a separate PR to avoid blocking on the spec merging.

**Comment by imphil** [comment]
Status update: a working OTBN toplevel, a top_earlgrey integration, a DIF and a hello world application is at https://github.com/imphil/opentitan/commits/bignum-init. I need to clean up some things and get some preliminary PRs merged, but this should already be a starting point for you @wallento.

Output from the OTBN hello world application which writes to OTBN IMEM and DMEM, and reads the DMEM back. The hardware currently only waits for a couple cycles and modifies DMEM during that time.

```
$ cat /dev/pts/6
I00000 boot_rom.c:35] Version:    opentitan-snapshot-20191101-1-1437-gfdc923f2
Build Date: 2020-06-17, 22:17:02

I00001 boot_rom.c:44] Boot ROM initialisation has completed, jump into flash!
I00000 hello_otbn.c:49] Hello OTBN!
I00001 hello_otbn.c:50] Built at: Jun 19 2020, 18:34:02
I00002 hello_otbn.c:52] Writing OTBN IMEM
I00003 hello_otbn.c:54] IMEM written
I00004 hello_otbn.c:59] Verifying IMEM through readback
I00005 hello_otbn.c:70] Writing OTBN DMEM
I00006 hello_otbn.c:72] DMEM written
I00007 hello_otbn.c:77] Verifying DMEM through readback
I00008 hello_otbn.c:88] Starting OTBN ...
I00009 hello_otbn.c:98] OTBN finished its operation
I00010 hello_otbn.c:100] Reading back OTBN DMEM after processing
I00011 hello_otbn.c:103] DMEM at 32b word 0 modified from 0x1234567 to 0x89abcdef
I00012 hello_otbn.c:103] DMEM at 32b word 1 modified from 0x89abcdef to 0x1234567
I00013 hello_otbn.c:103] DMEM at 32b word 2 modified from 0xdeadbeef to 0x89abcdef
I00014 hello_otbn.c:103] DMEM at 32b word 3 modified from 0x42424242 to 0x1234567
I00015 hello_otbn.c:103] DMEM at 32b word 4 modified from 0xdeadbeef to 0x89abcdef
I00016 hello_otbn.c:103] DMEM at 32b word 5 modified from 0x42424242 to 0x1234567
I00017 hello_otbn.c:103] DMEM at 32b word 6 modified from 0x1234567 to 0x89abcdef
I00018 hello_otbn.c:103] DMEM at 32b word 7 modified from 0x89abcdef to 0x1234567
```

**Comment by wallento** [comment]
Nice, thanks, will pick it up as starting point today

**Comment by imphil** [comment]
Implemented and merged in #2641
